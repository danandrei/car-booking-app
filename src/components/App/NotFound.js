import React from 'react';

const NotFound = () => (
  <div className="text-center cars-wrap">
    <h1>404! Not found.</h1>
  </div>
);

export default NotFound;
