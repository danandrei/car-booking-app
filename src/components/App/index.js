import React, { Component } from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';

import { history } from '../../helpers'
import { userActions } from '../../actions';
import {
  customerIsNotAuthenticatedRedir,
  adminIsNotAuthenticatedRedir,
} from '../../helpers'
import '../../assets/styles/index.scss';
import Navbar from '../Navbar';
import CarsPage from '../CarsPage';
import CarPage from '../CarPage';
import HomePage from '../HomePage';
import NotFound from './NotFound';
import BookingPage from '../BookingPage';

class App extends Component {

  render() {
    return (
      <div className="h-100">
        <Navbar />
          <Router history={history}>
            <Switch>
              <Route exact path="/" component={HomePage} />
              <Route path="/book" component={customerIsNotAuthenticatedRedir(BookingPage)} />
              <Route exact path="/cars" component={adminIsNotAuthenticatedRedir(CarsPage)} />
              <Route path="/cars/:id" component={adminIsNotAuthenticatedRedir(CarPage)} />
              <Route component={NotFound} />
            </Switch>
          </Router>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};

export default connect(mapStateToProps, {
  getCurrentUser: userActions.getCurrentUser,
})(App);
