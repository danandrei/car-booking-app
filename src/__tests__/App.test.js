/* MOCKS */
jest.mock('../components/Navbar', () => () => (
  <h1>navbar</h1>
));
jest.mock('../components/CarsPage', () => () => (
  <h1>cars page</h1>
));
jest.mock('../components/CarPage', () => () => (
  <h1>car page</h1>
));
jest.mock('../components/App/NotFound', () => () => (
  <h1>not found</h1>
));
jest.mock('../components/HomePage', () => () => (
  <h1>home page</h1>
));
jest.mock('../components/BookingPage', () => () => (
  <h1>booking page</h1>
));

// Dependencies
import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import App from '../components/App';

describe('>>>App--- Snapshot',()=>{
    it('+++capturing Snapshot of App', () => {
        const renderedValue =  renderer.create(<App />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
