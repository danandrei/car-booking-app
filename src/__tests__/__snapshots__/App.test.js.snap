// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>App--- Snapshot +++capturing Snapshot of App 1`] = `
<div>
  <h1>
    navbar
  </h1>
  <h1>
    App
  </h1>
</div>
`;
